pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -336,6 +340,18 @@ func (r *ReconcileHyperConverged) doReconcile(req *common.HcoRequest) (reconcile
 	knownHcoVersion, _ := req.Instance.Status.GetVersion(hcoVersionName)
 
 	if !r.upgradeMode && !init && knownHcoVersion != r.ownVersion {
+		if hcoutil.GetClusterInfo().IsManagedByOLM() {
```

Comment:
Sicnse most of the values are constant, lets add it all to the `setOperatorCondition` method and get a cleaner code.